core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1000,33 +1001,81 @@ private[spark] object SparkSubmitUtils {
     "tags_", "unsafe_")
 
   /**
-   * Represents a Maven Coordinate
-   * @param groupId the groupId of the coordinate
-   * @param artifactId the artifactId of the coordinate
-   * @param version the version of the coordinate
+   * Represents a Artifact Coordinate to resolve
+   * @param groupId the groupId of the artifact
+   * @param artifactId the artifactId of the artifact
+   * @param version the version of the artifact
+   * @param extraParams the extra params to resolve the artifact
    */
-  private[deploy] case class MavenCoordinate(groupId: String, artifactId: String, version: String) {
-    override def toString: String = s"$groupId:$artifactId:$version"
+  private[deploy] case class MavenCoordinate(groupId: String, artifactId: String, version: String,
+    extraParams: Map[String, String] = Map()) {
+
+    def params: String = if (extraParams.isEmpty) {
+        ""
+      } else {
+        "?" + extraParams.map{ case (k, v) => (k + "=" + v) }.mkString(":")
+      }
+
+    override def toString: String = s"$groupId:$artifactId:$version$params"
   }
 
   /**
-   * Extracts maven coordinates from a comma-delimited string. Coordinates should be provided
-   * in the format `groupId:artifactId:version` or `groupId/artifactId:version`.
-   * @param coordinates Comma-delimited string of maven coordinates
-   * @return Sequence of Maven coordinates
+   * Extracts artifact coordinates from a comma-delimited string. Coordinates should be provided
+   * in the format `groupId:artifactId:version?param1=value1\&param2\&value2:..` or
+   * `groupId/artifactId:version?param1=value1\&param2=value2:..`
+   *
+   * Param splitter & is the background process char in cli, so when multiple params is used,
```

Comment:
I don't think this comment is clear. "Shells usually use `&` as a reserved character, so escape these characters with `\&` or double-quote the whole argument if used"?